src/session/sessionDrawer.component.tsx

```diff
@@ -23,20 +23,30 @@ const StyledDrawer = styled(Drawer)(({ theme }) => ({
 
 export interface SessionDrawerProps {
   openSessionSave: () => void;
+  openSessionEdit: (sessionData: SessionListItem) => void;
+  openSessionDelete: (sessionData: SessionListItem) => void;
   sessionsList: SessionListItem[] | undefined;
   loadedSessionId: string | undefined;
-  onChangeLoadedSessionId: (selectedSessionId: string | undefined) => void;
+  onChangeLoadedSessionId: (loadedSessionId: string | undefined) => void;
 }
 
 interface SessionListElementProps extends SessionListItem {
   handleImport: (sessionId: string) => void;
   selected: boolean;
+  openSessionEdit: (sessionData: SessionListItem) => void;
+  openSessionDelete: (sessionData: SessionListItem) => void;
 }
 
 const SessionListElement = (
   props: SessionListElementProps
 ): React.ReactElement => {
-  const { selected, handleImport, ...session } = props;
+  const {
+    openSessionDelete,
+    openSessionEdit,
+    selected,
+    handleImport,
+    ...session
+  } = props;
 
   return (
     <Box
```

Comment:
Can we refactor this component to actually use a `List`, and to use the correct `ListItem` components like `ListItemButton`, `ListItemText`. It means we have to do less custom styling (e.g. ListItemButton already has a selected prop) and ensures we're using the correct semantic HTML/accessibility stuff. I have an example of the refactor that I can commit if you'd like :)

Reply:
Yeah can you commit the example please :)